src/useDitto.tsx

```diff
@@ -0,0 +1,39 @@
+import { Ditto } from "@dittolive/ditto";
+import { useContext, useState, useEffect } from "react";
+import { DittoHash, RegisterDitto, UnregisterDitto, DittoContext } from ".";
+
+export const useDitto = (
+  path?: string
+): {
+  ditto: Ditto | undefined;
+  dittoHash: DittoHash;
+  registerDitto?: RegisterDitto;
+  unregisterDitto?: UnregisterDitto;
+} => {
+  const { dittoHash, registerDitto, unregisterDitto } =
+    useContext(DittoContext);
+    
+  const [ditto, setDitto] = useState<Ditto | undefined>()
+
+  useEffect(() => {
+    let foundDitto: Ditto;
+    if (path) {
+      foundDitto = dittoHash[path];
+    } else {
+      const [first] = Object.values(dittoHash);
+      foundDitto = first;
+    }
+    if (foundDitto) {
+      setDitto(foundDitto);  
+    } else {
+      setDitto(undefined);
+    }
+  }, [path, dittoHash]);
+
+  return {
+    ditto,
+    dittoHash,
+    registerDitto,
+    unregisterDitto,
```

Comment:
I'm not 100% sure that it's a good idea to allow child nodes of the context provider to register and unregister Ditto instances from the provider. This could lead to sibling nodes losing references to a Ditto instance. Perhaps it's best to leave these functions out of the `useDitto` hook, such that the hook simply returns a Ditto instance?